server/src/main/java/io/crate/replication/logical/ShardReplicationChangesTracker.java

```diff
@@ -363,15 +365,14 @@ public void close() throws IOException {
         if (cancellable != null) {
             cancellable.cancel();
         }
-        shardReplicationService.getRemoteClusterClient(
-            shardId.getIndex(),
-            client -> RetentionLeaseHelper.attemptRetentionLeaseRemoval(
-                shardId,
-                clusterName,
-                client,
-                ActionListener.wrap(() -> {})
-            ),
-            e -> {}
-        );
+        shardReplicationService.getRemoteClusterClient(shardId.getIndex())
+            .whenComplete((client, err) -> {
+                RetentionLeaseHelper.attemptRetentionLeaseRemoval(
```

Comment:
same here, shouldn't we only renew if no error happened?

Reply:
Changed to use `thenAccept`